src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -267,7 +267,18 @@ ExecuteDistributedModify(Task *task)
 		result = PQexec(connection, task->queryString);
 		if (PQresultStatus(result) != PGRES_COMMAND_OK)
 		{
-			ReportRemoteError(connection, result);
+			int errorCode = ReportRemoteError(connection, result);
+
+			/* check if the error code is in integrity constraint violation class */
+			if (ERRCODE_TO_CATEGORY(ERRCODE_INTEGRITY_CONSTRAINT_VIOLATION) ==
+				ERRCODE_TO_CATEGORY(errorCode))
+			{
+				ereport(ERROR, (errcode(errorCode),
+								errmsg("integrity constraint violation on the "
+									   "worker node"),
+								errhint("Check the warning message above for details.")));
```

Comment:
I don't like log statements that ask the user to look for other log statements. We should just raise the error once with all the fields the user needs, not warn once (with the text the user probably wants) and error a second time (with the code, but not text, needed by the user).

As many drivers will expose this error text to the programming environment using them, asking the user to check previous logs isn't really tenable.